src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([])
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const fetchAllData = () => {
+    getAll().then(setGoods);
+  }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const fetchRedData = () => {
+    getRedGoods().then(setGoods);
+  }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const fetchFiveData = () => {
+    get5First().then(setGoods);
+  }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  useEffect(() => {
+    fetchAllData();
+  }, [])
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+       type="button"
+       data-cy="all-button"
+       onClick={fetchAllData}
```

Comment:
```suggestion
        type="button"
        data-cy="all-button"
        onClick={fetchAllData}
```

Consider fixing indentations accordingly to the code style guide